test_runner/fixtures/utils.py

```diff
@@ -647,6 +649,24 @@ def allpairs_versions():
     return {"argnames": "combination", "argvalues": tuple(argvalues), "ids": ids}
 
 
+def size_to_bytes(hr_size: str) -> int:
+    """
+    Gets human-readable size from postgresql.conf (e.g. 512kB, 10MB)
+    returns size in bytes
+    """
+    M_PREFIXES = {"": 1, "k": 1024, "M": 1024**2, "G": 1024**3, "T": 1024**4, "P": 1024**5}
+    match = re.search(r"^\'?(\d+)\s*([kMGTP]?)(B?)\'?$", hr_size)
```

Comment:
I'd do something similar to what we have in `humantime_to_ms`:
- Add `B` to `M_PREFIXES` (should it be `units` or `unit_multiplier_map` as in `humantime_to_ms`?)
- Use `({'|'.join(unit_multiplier_map.keys())})`, then we won't need to have a separate `b` variable

https://github.com/neondatabase/neon/blob/338ddd6d4075b25d004d3a631f6ba3e3a22ddff4/test_runner/fixtures/utils.py#L458-L466

Nit: the current regex matches unbalanced quotes like `'1MB` or `1MB'`, feel free to ignore this :)

Reply:
Nice catch about B in prefixes, it can make the code simpler, thanks. It allows us to get rid of `b` variable as well.
Regarding generating regex from the hashmap, I agree that it's a smart solution, but I believe that in this case the construction `([kMGTP]?B)?` is readable enough and slightly shorter.